Tech debt: create a pegasusCommon entry point

caleybrock opened the pull request:
Follow up on: https://github.com/code-dot-org/code-dot-org/pull/14612
- Remove duplicated browser-detector.js file from shared
- Adds hamburger.js to pegasusCommon.js and code-studio.js
- Removes hamburger.js as an entry point

Bjvanminnen commented on pegasus/sites.v3/code.org/layouts/default.haml:
```diff
@@ -1,4 +1,5 @@
 %link{:href=>"/shared/css/warning-banner.css", :rel=>"stylesheet"}
+%script{src: minifiable_asset_path("js/pegasusCommon.js")}
```
Instead of calling this pegasusCommon and having to silence it, could we create two version (that end up requiring the same code), and have each of them match the pegasus haml file names where they're being used?

caleybrock replied:
We have a few places where this gets used, so we'd start making lots of entry points which doesn't seem ideal. (code.org and hourofcode.com don't have single top level haml files). 
@pcardune do you have strong opinions on this?

caleybrock replied:
Actually @breville there might be a more common place to put this [theme_common_header_after](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/views/theme_common_head_after.haml)?
I also don't understand the difference between /layouts and /themes in pegasus. Is it true that one of each is always used on a page? Right now I put all this js in layouts, but that might be wrong.

breville replied:
Honestly the separation between themes and layouts does seem muddy and I'm not sure how it got to that point (a number of people had different philosophies about factoring in pegasus page construction).  But basically we start with the page body itself, wrap that in the layout, and then wrap that in the theme.  (The `= body` in the HAML for layout and theme is how they wrap their content.)  But both theme and layout are optional, so it comes down to how the pages are constructed.  https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/router.rb#L226-L239

From eyeballing the layouts and themes in use for code.org, it does appear that `theme_common_header_after` could be included in all the right places.

breville commented on apps/src/hamburger/hamburger.js:
```diff
@@ -1,40 +1,44 @@
-    $('#hamburger ul').slideToggle();
-    e.preventDefault();
-  });
+export const setupHamburgerMenu = function () {
```
initHamburger?

breville commented on apps/Gruntfile.js:
```diff
@@ -450,8 +450,7 @@ testsContext.keys().forEach(testsContext);
 
-    // Hamburger for header on dashboard & pegasus.
-    hamburger: './src/hamburger/hamburger.js',
+    pegasusCommon: './src/sites/code.org/pages/pegasusCommon.js',
```
makes sense being in pages?

caleybrock commented on pegasus/sites.v3/code.org/views/mobile_header_responsive.haml:
```diff
@@ -1,5 +1,4 @@
 %link{:href=>'/shared/css/hamburger.css', :rel=>'stylesheet'}
-%script{src: minifiable_asset_path("js/hamburger.js")}
```
potentially create an error here to log if we missed something

caleybrock commented:
Update here. I chatted with Tanya about top level entry points for pegasus and she said that there is a theme on each page and each theme uses `theme_common_body_after.haml`
I created two entry points, one for code.org and one for hoc.com and added it to the haml files in each of the respective directories. 
Brent and I chatted about adding a js snippet that could log a js error to new relic where we removed our script tags so we would know if it ever doesn't get included. I feel good about this solution now and don't think that's necessary. 

The only thing is I can't figure out how to not add the entry point to `checkEntryPoints` - if I don't it complains I didn't add the entryPoints to /sites, even though that's where they live.

Bjvanminnen commented:
You're including from haml file `pegasus/sites.v3/code.org/views/theme_common_head_after.haml` so I think you need your filename to be `sites/code.org/views/theme_common_header_after.js`
